Cargo.toml

```diff
@@ -14,3 +14,4 @@ categories = ["compression"]
 edition = "2018"
 
 [dependencies]
+thiserror = "1.0.29"
```

Comment:
Hmm, I'm not entirely certain that this dependency pulls its own weight. It brings `proc-macro2`, `quote` and `syn`, which are quite "heavy", and we only use a single time… I think I would rather keep the manual `impl`.

Reply:
Hmm - yeah that is a good point. Sure, let me roll back the latest commit.